backend/cmm_helpers.ml

```diff
@@ -2142,6 +2223,10 @@ let direct_apply lbl ty args (pos, _mode) dbg =
   Cop (Capply (ty, pos), Cconst_symbol (lbl, dbg) :: args, dbg)
 
 let call_caml_apply ty args_type mut clos args pos mode dbg =
+  let args_type =
+    List.map Extended_machtype.change_tagged_int_to_val args_type
+  in
+  let ty = Extended_machtype.change_tagged_int_to_val ty in
```

Comment:
Don't we want to preserve the fact that args and return are tagged integers to avoid having them scanned by the GC below, and only use `change_tagged_int_to_val` when calling `apply_function_sym`?

Reply:
Fixed